src/core/src/command.rs

```diff
@@ -50,38 +50,38 @@ impl<B: Backend> Submit<B> {
 ///
 /// Pools will always return an Encoder on `acquire_command_buffer` to provide a safe interface.
 #[derive(Debug)]
-pub struct Encoder<'a, B: Backend, C: CommandBuffer<B> + 'a>(&'a mut C, PhantomData<B>);
+pub struct Encoder<B: Backend, C: CommandBuffer<B>>(C, PhantomData<B>);
```

Comment:
oh wow, pretty! I think we should not specify `C` here though (even shorter!):
```rust
pub struct Encoder<B, C>(C, PhantomData<B>);
```
I know it may sound controversial. On one hand, this version is shorter and allows any higher-level wrappers to be shorter too. On the other hand, specifying the dependencies here allows for earlier error catching and clearer error messages. That point is rather weak though, and I recall the recommendation (not to use the bounds on struct definitions) to be in the guidelines.